File: Task-03/Ruby/Sub-Task 4/Pattern-FileHandling2.rb
n = File.read('input.txt').to_i

File.open('output.txt', 'w') do |file|
  n.times {|i|file.puts' '*(n-i-1) + '*'*(2*i+1)}
  (n-1).times {|i|file.puts' ' *(i+1) + '*'*(2*(n-i-2)+1)}
end

File: Task-03/Elixir/Sub-Task 4/Pattern-FileHandling2.ex
n = String.to_integer(File.read!("input.txt"))

File.write!("output.txt", Enum.join(
  for i <- 0..(n-1), do: String.duplicate(" ", n-i-1) <> String.duplicate("*", 2*i+1) <> "\n"
) <> Enum.join(
  for i <- (n-2)..0, do: String.duplicate(" ", n-i-1) <> String.duplicate("*", 2*i+1) <> "\n"
))
